velox/functions/prestosql/tests/DecimalArithmeticTest.cpp

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <optional>
+#include "velox/functions/prestosql/tests/FunctionBaseTest.h"
+#include "velox/vector/tests/VectorTestBase.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+using namespace facebook::velox::functions::test;
+
+namespace facebook::velox {
+
+class DecimalArithmeticTest : public FunctionBaseTest {
+ protected:
+  template <typename T>
+  void testDecimalExpr(
+      const VectorPtr& expected,
+      const std::string& expression,
+      const std::vector<VectorPtr>& input) {
+    auto result = evaluate<SimpleVector<T>>(expression, makeRowVector(input));
+    assertEqualVectors(expected, result);
+    ASSERT_EQ(result->type()->toString(), expected->type()->toString());
+  }
+
+  template <typename T>
+  void testOpFlatConstant(
+      const std::string& operation,
+      const std::string& constant,
+      const VectorPtr& flatVector,
+      const VectorPtr& result,
+      bool isLeftConstant) {
+    auto shortConstant = BaseVector::wrapInConstant(
+        1, 0, makeDecimalFlatVector<ShortDecimal>({ShortDecimal(1000)}, 10, 3));
+    std::vector<VectorPtr> input({flatVector});
+    RowVectorPtr rowVector = makeRowVector(input);
+    auto rowType = std::dynamic_pointer_cast<const RowType>(rowVector->type());
+    core::TypedExprPtr typedExpr;
```

Comment:
As discussed, let's add support for parsing decimal literals.

Reply:
Will work on parsing decimal literals in another PR. The first step is to support duck variant to velox variant conversion. A PR is in-progress: https://github.com/facebookincubator/velox/pull/2006